pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -454,17 +457,30 @@ def urlpattern(*args, **kwargs):
 
 
 class CollectionUploadViewSet(
-    ExceptionHandlerMixin, viewsets.GenericViewSet, UploadGalaxyCollectionMixin
+    ExceptionHandlerMixin, UploadGalaxyCollectionMixin, SingleArtifactContentUploadViewSet
 ):
     """
     ViewSet for Collection Uploads.
     """
 
-    serializer_class = CollectionOneShotSerializer
+    queryset = CollectionVersion.objects.all()
+    serializer_class = CollectionVersionUploadSerializer
     pulp_tag_name = "Pulp_Ansible: Artifacts Collections V3"
 
     DEFAULT_ACCESS_POLICY = _PERMISSIVE_ACCESS_POLICY
 
+    _declared_fields = serializer_class._declared_fields
+    INLINE_SERIALIZER = inline_serializer(
```

Comment:
So I can hide the Pulp-specific fields: `artifact`, `upload`, and `repository` from the generated schema. I tried to just not include them in the `Meta` class, but then it would error when `artifact` or `repository` was set by the viewset's logic.

Reply:
It think, this is dangerous. Just not adding them to the docs, means that the serializer still accepts them. And it shouldn't!?

Reply:
You are right. I'll add validation for this serializer.